server/api/router/post.ts

```diff
@@ -187,10 +187,27 @@ export const postRouter = createTRPCRouter({
         });
       }
 
-      const [deletedPost] = await ctx.db
-        .delete(post)
-        .where(eq(post.id, id))
-        .returning();
+      const deletedPost = await ctx.db.transaction(async (tx) => {
```

Comment:
Nitpick, naming here should be `ctx`

Reply:
would txCtx be a good middle ground?

ie. its the transaction context

Reply:
@JohnAllenTech This looks good to me.
Let me know if that works.